middle_end/flambda/types/env/binding_time.ml

```diff
@@ -68,6 +69,14 @@ module With_name_mode = struct
     | 2 -> Name_mode.phantom
     | _ -> assert false
 
+  let scoped_name_mode t ~min_binding_time =
+    if strictly_earlier (binding_time t) ~than:earliest_var
+    || strictly_earlier min_binding_time ~than:(binding_time t)
```

Comment:
This second check doesn't seem right to me: I was expecting it to allow `binding_time t` equal to `min_binding_time`, given the name `min_binding_time`.

Reply:
I agree, but the Binding_time module only exposed the `strictly_earlier` comparison and I chose to ensure that `min_binding_time` is never equal to the binding time of a variable to get around the issue (this part of the code used to be in `Typing_env`). I could probably relax that restriction to allow binding times equal to `min_binding_time` now that I'm confident that the rest of the code is correct.